accounts-db/src/accounts_index/bucket_map_holder.rs

```diff
@@ -209,7 +250,32 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> BucketMapHolder<T, U>
 
         let disk = match config.index_limit {
             IndexLimit::InMemOnly => None,
-            IndexLimit::Minimal => Some(BucketMap::new(bucket_config)),
+            IndexLimit::Minimal | IndexLimit::Threshold(_) => Some(BucketMap::new(bucket_config)),
+        };
+
+        // Compute threshold_entries once here
+        let threshold_entries_per_bin = match config.index_limit {
+            IndexLimit::InMemOnly | IndexLimit::Minimal => None,
+            IndexLimit::Threshold(limit_bytes) => {
+                let bytes_per_bin = (limit_bytes as usize) / bins;
+                let entries_per_bin = bytes_per_bin
+                    / (InMemAccountsIndex::<T, U>::size_of_uninitialized()
+                        + InMemAccountsIndex::<T, U>::size_of_single_entry());
+                let target_entries_per_bin =
+                    Self::calculate_target_entries_per_bin(entries_per_bin);
+                let threshold_entries_per_bin =
+                    Self::calculate_threshold_entries(target_entries_per_bin);
+
+                info!(
+                    "AccountsIndex threshold configured: limit_bytes={limit_bytes}, \
+                     bytes_per_bin={bytes_per_bin}, entries_per_bin={entries_per_bin}, \
+                     target_entries_per_bin={}, high_water_mark={}, low_water_mark={}",
+                    threshold_entries_per_bin.target_entries,
+                    threshold_entries_per_bin.high_water_mark,
+                    threshold_entries_per_bin.low_water_mark,
+                );
```

Comment:
Can we also log the target bytes? This would be the target_entries_per_bin * bins * bytes_per_entry.


```suggestion
                info!(
                    "AccountsIndex threshold configured: limit_bytes={limit_bytes}, target_bytes={}, \
                     bytes_per_bin={bytes_per_bin}, entries_per_bin={entries_per_bin}, \
                     target_entries_per_bin={}, high_water_mark={}, low_water_mark={}",
                    target_entries_per_bin * bins * bytes_per_entry,
                    threshold_entries_per_bin.target_entries,
                    threshold_entries_per_bin.high_water_mark,
                    threshold_entries_per_bin.low_water_mark,
                );
```